lib/pos_leader_schedule.go

```diff
@@ -7,7 +7,7 @@ import (
 	"github.com/pkg/errors"
 )
 
-func (bav *UtxoView) GenerateLeaderSchedule() ([]*PKID, error) {
+func (bav *UtxoView) GenerateLeaderSchedule(maxValidatorSet []*ValidatorEntry) ([]*PKID, error) {
```

Comment:
nit: in all other parts of the code, validatorSet already refers to the full validator set, so there's no need to disambiguate that here
```suggestion
func (bav *UtxoView) GenerateLeaderSchedule(validatorSet []*ValidatorEntry) ([]*PKID, error) {
```